RxCocoa/Foundation/NSObject+Rx+RawRepresentable.swift

```diff
@@ -27,6 +27,40 @@ extension Reactive where Base: NSObject {
         return observe(E.RawValue.KVOType.self, keyPath, options: options, retainSelf: retainSelf)
             .map(E.init)
     }
+
+    /**
+     Specialization of generic `observe` method.
+
+     This specialization first observes `KVORepresentable` value and then converts it to `RawRepresentable` value.
+
+     It is useful for observing bridged ObjC enum values.
+
+     For more information take a look at `observe` method.
+     */
+    @available(swift 4.0)
+    public func observe<E: RawRepresentable>(_ keyPath: KeyPath<Base, E>, options: KeyValueObservingOptions = [.new, .initial], retainSelf: Bool = true) -> Observable<E?> where E.RawValue: KVORepresentable {
```

Comment:
It seems to me that the result should be `Observable<E?>`.

We could do something like.

```
return observe(E.self, keyPathString, options: options, retainSelf: retainSelf).flatMap { value in
    guard let value  = value else { 
         #if DEBUG
             rxFatalError("Something went wrong with KVO observing mechanism")
         #endif
         return Observable.empty()
    }
    return 
}
```

The reason why I wouldn't want to make this a fatal error in release mode is because ObjC <-> Swift interoperability is a bit flakey.